pkg/plugin/integration_test.go

```diff
@@ -744,3 +751,176 @@ func TestHowActionsCauseRotations(t *testing.T) {
 		})
 	}
 }
+
+func fileHash(path string) (string, error) {
+	hasher := sha256.New()
+	s, err := ioutil.ReadFile(path)
+	hasher.Write(s)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(hasher.Sum(nil)), nil
+}
+
+func hashCurrentFiles(az *fakecloud.AzureCloud, log *logrus.Entry) func() map[string]map[string][]string {
+	type bothHashes struct {
+		before string
+		after  string
+	}
+	hashes := map[string]bothHashes{}
+	prefix := ""
+	parentDir := ""
+	beforeCallback := func(path string, fi os.FileInfo, err error) error {
+		if fi.IsDir() {
+			return nil
+		}
+		hash, err := fileHash(path)
+		if err != nil {
+			return err
+		}
+		hashes[prefix+strings.TrimPrefix(path, parentDir)] = bothHashes{before: hash}
+		return nil
+	}
+	afterCallback := func(path string, fi os.FileInfo, err error) error {
+		if fi.IsDir() {
+			return nil
+		}
+		hash, err := fileHash(path)
+		if err != nil {
+			return err
+		}
+		var bh bothHashes
+		bh, ok := hashes[prefix+strings.TrimPrefix(path, parentDir)]
+		if !ok {
+			bh = bothHashes{}
+		}
+		bh.after = hash
+		hashes[prefix+strings.TrimPrefix(path, parentDir)] = bh
+		return nil
+	}
+	for _, scaleset := range az.ComputeRP.State {
+		for _, vm := range scaleset.Vms {
+			prefix = fmt.Sprintf("%s:%s:", scaleset.Name, *vm.InstanceID)
+			parentDir = scaleset.VmsDir[*vm.InstanceID]
+			filepath.Walk(scaleset.VmsDir[*vm.InstanceID], beforeCallback)
+		}
+	}
+
+	return func() map[string]map[string][]string {
```

Comment:
I find the program flow here to be pretty confusing---we defined two local callbacks already, and here we return another function, to be invoked at some later time.

I think this would be easier for me to follow (and we'd gain a little more code reusability) if we wrote a single function that returned a map of hashes of all the files from a scaleset. Then, we could run that in our tests at specified before and after moments, letting the caller keep track of that state rather than trying to do that for them here. And rather than embedding the diffing logic here, we could split that out into its own helper.